pkg/ifaces/registerer.go

```diff
@@ -55,19 +71,54 @@ func (r *Registerer) Subscribe(ctx context.Context) (<-chan Event, error) {
 // IfaceNameForIndex gets the interface name given an index as recorded by the underlying
 // interfaces' informer. It backs up into the net.InterfaceByIndex function if the interface
 // has not been previously registered
-func (r *Registerer) IfaceNameForIndex(idx int) (string, bool) {
+func (r *Registerer) IfaceNameForIndexAndMAC(idx int, mac [6]uint8) (string, bool) {
 	r.m.RLock()
-	name, ok := r.ifaces[idx]
+	macsMap, ok := r.ifaces[idx]
 	r.m.RUnlock()
-	if !ok {
-		iface, err := net.InterfaceByIndex(idx)
-		if err != nil {
-			return "", false
+	if ok {
+		if len(macsMap) == 1 {
+			// No risk of collision, just return entry without checking for MAC
+			for _, name := range macsMap {
+				return name, true
+			}
+		} else if len(macsMap) > 1 {
+			// Several entries, need to disambiguate by MAC
+			name, ok := macsMap[mac]
+			if ok {
+				return name, true
+			}
+			// Not found => before falling back to syscall lookup that is CPU intensive, run this quick ovn optimization:
+			// eth0 & ens5 often collide; MAC starting with 0A:58 should be eth0 and not ens5, but since the MAC captured in flow
+			// doesn't match the actual interface MAC, we'll hardcode that.
+			if mac[0] == 0x0a && mac[1] == 0x58 {
```

Comment:
I don't think this an endian neutral way comparing mac this way ?

Reply:
I'm not super knowledgeable on endianess by I would say endianess doesn't affect byte arrays? It matters for bigger-than-byte types, like ints, but with a byte array _we_ control what's on the lower and upper index, this is not left to the architecture I think? E.g. an int16 `0x1234` could be `[0x12, 0x34]` or `[0x34, 0x12]` but a byte array `[0x12, 0x34]` always stays as is? I guess there could be an issue if macs were provided as ints but afaik it's never the case.
Again, I'm not super confident on this topic ...

Reply:
> case.

Just take note of this to be verified on s390x as it's the only BE arch we have atm